libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/WebexAdapter.cs

```diff
@@ -0,0 +1,350 @@
+
+            botkit.Ready(() => { (botkit.Adapter as WebexAdapter).RegisterWebhookSubscription(botkit.GetConfig("webhook_uri").ToString()); });
+        }
+
+        /// <summary>
+        /// Clears out and resets all the webhook subscriptions currently associated with this application.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<object> ResetWebhookSubscriptions()
+        {
+            var task = Task.Run(() => this.api.ListWebhooksAsync();
+            task.ContinueWith((list) => 
+            {
+                for (int i = 0; i <  list.Result.)
+                {
+                    //await this.api.DeleteWebhookAsync(list.items[i]);
+                }
+            });
+        }
+
+        /// <summary>
+        /// Register a webhook subscription with Webex Teams to start receiving message events.
+        /// </summary>
+        /// <param name="webhookPath">The path of the webhook endpoint like `/api/messages`.</param>
+        public void RegisterWebhookSubscription(string webhookPath)
+        {
+            var webHookName = (this.config.WebhookName != null) ? this.config.WebhookName : "Botkit Firehose";
+
+            /*this.api.ListWebhooksAsync().ContinueWith((list) => {
+                string hookId = null;
+
+                for (int i; i < list items.Length; i++)
+                {
+
+                }
+            } );*/
+        }
+
+        /// <summary>
+        /// Standard BotBuilder adapter method to send a message from the bot to the messaging API.
+        /// </summary>
+        /// <param name="turnContext">A TurnContext representing the current incoming message and environment.</param>
+        /// <param name="activities">An array of outgoing activities to be sent back to the messaging API.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken)
+        {
+            List<ResourceResponse> responses = new List<ResourceResponse>();
+            for (int i = 0; i < activities.Length; i++)
+            {
+                var activity = activities[i];
+                if (activity.Type.Equals(ActivityTypes.Message))
+                {
+                    // transform activity into the webex message format
+                    var personIDorEmail = ((activity.ChannelData as dynamic).toPersonEmail != null) ? (activity.ChannelData as dynamic).toPersonEmail : activity.Recipient.Id;
+                    var text = (activity.ChannelData != null) ? (activity.ChannelData as dynamic).markdown : activity.Text;
+                    Message webexResponse = await this.api.CreateDirectMessageAsync(personIDorEmail, text);
```

Comment:
Why are you using CreateDirectMessageAsync instead of CreateMessageAsync?

Reply:
Because of the method signature being quite more similar to what we do in JS. I'm open to discuss this anyway